pkg/endpoint/bpf.go

```diff
@@ -59,7 +58,7 @@ func (e *Endpoint) PolicyMapPathLocked() string {
 
 // CallsMapPathLocked returns the path to cilium tail calls map of an endpoint.
 func (e *Endpoint) CallsMapPathLocked() string {
-	return bpf.LocalMapPath(loader.CallsMapName, e.ID)
+	return e.owner.Datapath().Loader().CallsMapPath(e.ID)
```

Comment:
Looking at these changes, I do wonder why the endpoint package needs to know the chain of objects from the owner down to figuring out the CallsMapPath. Though that said for this particular case I'd imagine we can get rid of these map paths helpers from endpoint in a next step of iteration on the endpoint datapath abstractions.

Reply:
Agreed - I think these refactors will review some odd dependency chains that we have which will motivate other refactors as well.